targets/TARGET_Atmel/TARGET_SAM_CortexM0P/TARGET_SAML21J18A/device/TOOLCHAIN_GCC_ARM/saml21j18a.ld

```diff
@@ -2,26 +2,42 @@ OUTPUT_FORMAT("elf32-littlearm", "elf32-littlearm", "elf32-littlearm")
 OUTPUT_ARCH(arm)
 SEARCH_DIR(.)
 
+#define ROM_START   0x00000000
+#define ROM_SIZE    0x00040000
+#define RAM_START   0x20000000
+#define RAM_SIZE    0x00008000
+#define VECTORS     45   /* This value must match NVIC_NUM_VECTORS */
+
+#if !defined(MBED_APP_START)
+  #define MBED_APP_START ROM_START
+#endif
+
+#if !defined(MBED_APP_SIZE)
+  #define MBED_APP_SIZE ROM_SIZE
+#endif
+
+#define VECTORS_SIZE (VECTORS * 4)
+
 /* Memory Spaces Definitions */
 MEMORY {
-    rom (rx)  : ORIGIN = 0x00000000, LENGTH = 0x00040000
-        ram (rwx) : ORIGIN = 0x20000000 + 0xB8, LENGTH = 0x00008000 - 0xB8
-    }
+    rom (rx)  : ORIGIN = MBED_APP_START, LENGTH = MBED_APP_SIZE
+    ram (rwx) : ORIGIN = RAM_START + VECTORS_SIZE, LENGTH = RAM_SIZE - VECTORS_SIZE
+}
 
-    /* The stack size used by the application. NOTE: you need to adjust according to your application. */
-    STACK_SIZE = DEFINED(STACK_SIZE) ? STACK_SIZE : DEFINED(__stack_size__) ? __stack_size__ : 0x2000;
+STACK_SIZE = DEFINED(STACK_SIZE) ? STACK_SIZE : DEFINED(__stack_size__) ? __stack_size__ : 4096;
```

Comment:
Does this need to be updated based on 5.12 stack unification (already on master) ?